parquet-scrooge/src/main/java/parquet/scrooge/ScroogeStructConverter.java

```diff
@@ -295,36 +317,56 @@ private List getEnumList(String enumName) throws ClassNotFoundException, Illegal
   }
 
   public ThriftType convertEnumTypeField(ThriftStructFieldInfo f) {
-    List<ThriftType.EnumValue> enumValues = new ArrayList<ThriftType.EnumValue>();
+    return convertEnumTypeField(f.manifest().runtimeClass(), f.tfield().name);
+  }
 
-    String enumName = f.manifest().runtimeClass().getName();
+  private ThriftType convertEnumTypeField(Class enumClass, String fieldName){
+    List<ThriftType.EnumValue> enumValues = new ArrayList<ThriftType.EnumValue>();
+    String enumName = enumClass.getName();
     try {
       List enumCollection = getEnumList(enumName);
       for (Object enumObj : enumCollection) {
-        ScroogeEnumDesc enumDesc = ScroogeEnumDesc.getEnumDesc(enumObj);
-        //be compatible with thrift generated enum which have capitalized name
+        ScroogeEnumDesc enumDesc = ScroogeEnumDesc.fromEnum(enumObj);
         enumValues.add(new ThriftType.EnumValue(enumDesc.id, enumDesc.originalName));
       }
       return new ThriftType.EnumType(enumValues);
-    } catch (ReflectiveOperationException e) {
-      throw new ScroogeSchemaConversionException("Can not convert enum field " + f, e);
+    } catch (Exception e) {
+      throw new ScroogeSchemaConversionException("Can not convert enum field " + fieldName, e);
+    }
+  }
+
+  private boolean isUnion(Class klass){
+    for(Field f: klass.getDeclaredFields()) {
+      if (f.getName().equals("Union"))
+        return true;
```

Comment:
I don't understand this logic.
Is it a union if one of the fields is called "Union"?